src/components/ListItem.js

```diff
@@ -55,26 +65,30 @@ const ListItem = ({ item, onDelete, token }) => {
   }
 
   return (
-    <li className={className}>
-      <span className="label" aria-hidden="true">
-        {purchaseNext}
-      </span>
-      {item.name}
-      <span className="screen-reader-only">
-        Next purchase in {item.next_purchase} days.
-      </span>
-      <button
-        className={isPurchased ? 'purchased' : 'not-purchased'}
-        onClick={onHandle}
-        disabled={isPurchased ? !null : null}
-      >
-        Purchase
-        <span className="screen-reader-only">{item.name}.</span>
-      </button>
-      <button onClick={onDelete}>
-        <img className="trash" src={trash} alt="delete item" />
-      </button>
-    </li>
+    <>
+      <li className={className}>
+        <span className="label" aria-hidden="true">
+          {purchaseNext}
+        </span>
+        {item.name}
+        <span className="screen-reader-only">
+          Next purchase in {item.next_purchase} days.
+        </span>
+        <button
+          className={isPurchased ? 'purchased' : 'not-purchased'}
+          onClick={onHandle}
+          disabled={isPurchased ? !null : null}
+        >
+          Purchase
+          <span className="screen-reader-only">{item.name}.</span>
+        </button>
+        <button onClick={onDelete}>
+          <img className="trash" src={trash} alt="delete item" />
+        </button>
+        <button onClick={handleShow}>View Details</button>
+      </li>
+      <TestModal item={item} show={show} handleClose={handleClose} />
```

Comment:
This is great! Nice way to show the item details. Another option using a different library might be [Drawer](https://material-ui.com/components/drawers/) from Material UI